### PR TITLE
FeatureToggles: remove open type declaration

### PR DESCRIPTION
**Pull request by ryantxu**
The current typescript definition for feature toggles allows the frontend to specify *any* string without setting it in the registry.  This is how flags worked before we had a registry, so we could not just break that in one go.

This PR changes the typescript definition so it requires values are defined in the registry.

This change identified one feature toggle `azLegacyTemplateVariables` that was used, but not included in the registry.

### CODE REVIEW

File: pkg/services/featuremgmt/toggles_gen_test.go
Comment (ryantxu) [inline pkg/services/featuremgmt/toggles_gen_test.go:46]: ^^ I added it to the ownerless section since I don't know who/if it is still real

Interesting that both unclaimed flags are for azure

File: pkg/services/featuremgmt/registry.go
Comment (ArturWierzbicki) [inline pkg/services/featuremgmt/registry.go:71]:
```suggestion
			State:       FeatureStateBeta,
			Owner:       grafanaPartnerPluginsSquad,
```
and 

```go
	grafanaPartnerPluginsSquad                  codeowner = "@grafana/partner-plugins"
```

this one was added in https://github.com/grafana/grafana/pull/52584 - it looks like it belongs to partner plugins squad as its for the Azure Monitor datasource 

cc @aangelisc @vickyyyyyyy

**Review by chri2547 — COMMENTED** [review]
Docs look fine to me. Thank you!

File: pkg/services/featuremgmt/registry.go
Comment (aangelisc) [reply pkg/services/featuremgmt/registry.go]: This flag can be removed now @ArturWierzbicki 😊

**Review by ArturWierzbicki — APPROVED** [review]

**Review by aangelisc — APPROVED** [review]
To avoid blocking this PR any further I'll remove the `azLegacyTemplateVariables` feature flag once this is merged 😊

**Pull request merged**
